symphony-bdk-core/src/main/java/com/symphony/bdk/core/activity/ActivityRegistry.java

```diff
@@ -28,9 +31,37 @@
   /** The bot session forwarded to command-based activities only */
   private final UserV2 botSession;
 
+  private final MessageService messageService;
+
+  private SlashCommand helpCommand;
+
   /** The Datafeed real-time events source, or Datafeed listener */
   private final Consumer<RealTimeEventListener> realTimeEventsSource;
 
+  public ActivityRegistry(UserV2 botSession, Consumer<RealTimeEventListener> realTimeEventsSource , MessageService messageService) {
+    this.botSession = botSession;
+    this.realTimeEventsSource = realTimeEventsSource;
+    this.messageService = messageService;
+
+    this.registerHelpCommand();
```

Comment:
Pass `MessageService` as `registerHelpCommand` parameter and remove from class properties

Reply:
furthermore to avoid the dependency between ActivityRegistry and MessageService at all I would try to call 
this.activityRegistry.register(new HelpCommand(messageService, activityRegistry)) from the SymphonyBdk constructor